In [29]:
import pandas as pd

In [30]:
df =  pd.read_csv("Diwali Sales Data.csv", encoding='unicode_escape')
df.head()

User_ID  Cust_name Product_ID Gender Age Group  Age  Marital_Status  \
0  1002903  Sanskriti  P00125942      F     26-35   28               0   
1  1000732     Kartik  P00110942      F     26-35   35               1   
2  1001990      Bindu  P00118542      F     26-35   35               1   
3  1001425     Sudevi  P00237842      M      0-17   16               0   
4  1000588       Joni  P00057942      M     26-35   28               1   

            State      Zone       Occupation Product_Category  Orders  \
0     Maharashtra   Western       Healthcare             Auto       1   
1  Andhra Pradesh  Southern             Govt             Auto       3   
2   Uttar Pradesh   Central       Automobile             Auto       3   
3       Karnataka  Southern     Construction             Auto       2   
4         Gujarat   Western  Food Processing             Auto       2   

    Amount  Status  unnamed1  
0  23952.0     NaN       NaN  
1  23934.0     NaN       NaN  
2  23924.0     NaN       NaN  
3  23912.0     NaN       NaN  
4  23877.0     NaN       NaN

In [31]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11251 entries, 0 to 11250
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   User_ID           11251 non-null  int64  
 1   Cust_name         11251 non-null  object 
 2   Product_ID        11251 non-null  object 
 3   Gender            11251 non-null  object 
 4   Age Group         11251 non-null  object 
 5   Age               11251 non-null  int64  
 6   Marital_Status    11251 non-null  int64  
 7   State             11251 non-null  object 
 8   Zone              11251 non-null  object 
 9   Occupation        11251 non-null  object 
 10  Product_Category  11251 non-null  object 
 11  Orders            11251 non-null  int64  
 12  Amount            11239 non-null  float64
 13  Status            0 non-null      float64
 14  unnamed1          0 non-null      float64
dtypes: float64(3), int64(4), object(8)
memory usage: 1.3+ MB


In [32]:
df.describe()

User_ID           Age  Marital_Status        Orders        Amount  \
count  1.125100e+04  11251.000000    11251.000000  11251.000000  11239.000000   
mean   1.003004e+06     35.421207        0.420318      2.489290   9453.610858   
std    1.716125e+03     12.754122        0.493632      1.115047   5222.355869   
min    1.000001e+06     12.000000        0.000000      1.000000    188.000000   
25%    1.001492e+06     27.000000        0.000000      1.500000   5443.000000   
50%    1.003065e+06     33.000000        0.000000      2.000000   8109.000000   
75%    1.004430e+06     43.000000        1.000000      3.000000  12675.000000   
max    1.006040e+06     92.000000        1.000000      4.000000  23952.000000   

       Status  unnamed1  
count     0.0       0.0  
mean      NaN       NaN  
std       NaN       NaN  
min       NaN       NaN  
25%       NaN       NaN  
50%       NaN       NaN  
75%       NaN       NaN  
max       NaN       NaN

In [33]:
df["Status"].unique()

array([nan])

In [34]:
df['unnamed1'].unique()

array([nan])

In [35]:
# all values of status column are null and unnamed1 column has also null so remove them
df.drop(columns=["Status","unnamed1"], inplace=True)

In [36]:
df.head()

User_ID  Cust_name Product_ID Gender Age Group  Age  Marital_Status  \
0  1002903  Sanskriti  P00125942      F     26-35   28               0   
1  1000732     Kartik  P00110942      F     26-35   35               1   
2  1001990      Bindu  P00118542      F     26-35   35               1   
3  1001425     Sudevi  P00237842      M      0-17   16               0   
4  1000588       Joni  P00057942      M     26-35   28               1   

            State      Zone       Occupation Product_Category  Orders   Amount  
0     Maharashtra   Western       Healthcare             Auto       1  23952.0  
1  Andhra Pradesh  Southern             Govt             Auto       3  23934.0  
2   Uttar Pradesh   Central       Automobile             Auto       3  23924.0  
3       Karnataka  Southern     Construction             Auto       2  23912.0  
4         Gujarat   Western  Food Processing             Auto       2  23877.0

In [37]:
# checking duplicates in data
df[df.duplicated() == True]

User_ID   Cust_name Product_ID Gender Age Group  Age  Marital_Status  \
19     1001883     Praneet  P00029842      M     51-55   54               1   
4404   1004725     Jackson  P00150842      F     36-45   37               1   
5703   1003208      Bowman  P00171642      F     26-35   31               0   
5908   1001260     Dheeraj  P00344042      M     26-35   28               0   
6173   1001325       Reese  P00111742      F     26-35   27               1   
8651   1000083        Gute  P00242842      M     26-35   35               0   
8941   1001476     Anudeep  P00036842      M     18-25   25               0   
10571  1004404  Rittenbach  P00150142      F     26-35   28               1   

               State      Zone    Occupation       Product_Category  Orders  \
19     Uttar Pradesh   Central   Hospitality                   Auto       1   
4404     Maharashtra   Western   Hospitality  Electronics & Gadgets       4   
5703           Bihar   Eastern   Agriculture  Electronics & Gadgets       4   
5908     Maharashtra   Western     IT Sector  Electronics & Gadgets       4   
6173         Gujarat   Western  Construction  Electronics & Gadgets       3   
8651   Uttar Pradesh   Central   Hospitality     Clothing & Apparel       3   
8941     Maharashtra   Western     IT Sector     Clothing & Apparel       4   
10571        Haryana  Northern      Aviation  Electronics & Gadgets       3   

        Amount  
19     23568.0  
4404    9859.0  
5703    8088.0  
5908    8015.0  
6173    7923.0  
8651    5345.0  
8941    5202.0  
10571   2304.0

In [38]:
duplicate = df[df.duplicated(keep='last')]
duplicate

User_ID   Cust_name Product_ID Gender Age Group  Age  Marital_Status  \
18     1001883     Praneet  P00029842      M     51-55   54               1   
4403   1004725     Jackson  P00150842      F     36-45   37               1   
5702   1003208      Bowman  P00171642      F     26-35   31               0   
5907   1001260     Dheeraj  P00344042      M     26-35   28               0   
6172   1001325       Reese  P00111742      F     26-35   27               1   
8650   1000083        Gute  P00242842      M     26-35   35               0   
8940   1001476     Anudeep  P00036842      M     18-25   25               0   
10570  1004404  Rittenbach  P00150142      F     26-35   28               1   

               State      Zone    Occupation       Product_Category  Orders  \
18     Uttar Pradesh   Central   Hospitality                   Auto       1   
4403     Maharashtra   Western   Hospitality  Electronics & Gadgets       4   
5702           Bihar   Eastern   Agriculture  Electronics & Gadgets       4   
5907     Maharashtra   Western     IT Sector  Electronics & Gadgets       4   
6172         Gujarat   Western  Construction  Electronics & Gadgets       3   
8650   Uttar Pradesh   Central   Hospitality     Clothing & Apparel       3   
8940     Maharashtra   Western     IT Sector     Clothing & Apparel       4   
10570        Haryana  Northern      Aviation  Electronics & Gadgets       3   

        Amount  
18     23568.0  
4403    9859.0  
5702    8088.0  
5907    8015.0  
6172    7923.0  
8650    5345.0  
8940    5202.0  
10570   2304.0

In [39]:
#dropping duplicated data
data = df.drop_duplicates()


In [40]:
data.duplicated().unique()

array([False])

In [41]:
# finding sales ratio between men and women
data['Gender'].value_counts(normalize=True) * 100 

F    69.714489
M    30.285511
Name: Gender, dtype: float64

### Observation: almost 70% productss are ordered by Women only and 30% productss are ordered by Men

In [42]:
# finding sales ratio between different age-groups
data['Age Group'].value_counts(normalize=True) * 100 

26-35    40.362892
36-45    20.323757
18-25    16.703727
46-50     8.778796
51-55     7.391266
55+       3.806813
0-17      2.632749
Name: Age Group, dtype: float64

### Observation: 40% - most products are ordered by age group 26-35 only..
### only 2.63% - least products are ordered by age group below 17yrs

In [53]:
# finding sales ratio between married and unmarried people
data['Marital_Status'].value_counts(normalize=True) * 100 

0    57.97385
1    42.02615
Name: Marital_Status, dtype: float64

### Obeservation: Unmarried people are ordering more frequently in comparison to married people


In [54]:
# Finding sales ratio between different states across india
data['State'].value_counts(normalize=True) * 100 

Uttar Pradesh       17.290759
Maharashtra         13.546207
Karnataka           11.607222
Delhi                9.846126
Madhya Pradesh       8.209553
Andhra Pradesh       7.222272
Himachal Pradesh     5.407809
Kerala               4.029174
Haryana              4.011385
Bihar                3.851285
Gujarat              3.806813
Jharkhand            3.379881
Uttarakhand          2.846215
Rajasthan            2.054612
Punjab               1.778885
Telangana            1.111803
Name: State, dtype: float64

### Observation: Uttar Pradesh is the state with the highest number of orders in this dataset followed by Maharastra and Karnataka

In [45]:
data['Zone'].value_counts(normalize=True) * 100 

Central     38.192653
Southern    23.970471
Western     17.353020
Northern    13.252691
Eastern      7.231166
Name: Zone, dtype: float64

### Central Zone has the highest number of orders in this dataset

In [46]:
data['Occupation'].value_counts(normalize=True) * 100 

IT Sector          14.106555
Healthcare         12.523348
Aviation           11.642800
Banking            10.130748
Govt                7.595837
Hospitality         6.243885
Media               5.665748
Automobile          5.034244
Chemical            4.820777
Lawyer              4.722939
Retail              4.456106
Food Processing     3.762341
Construction        3.673397
Textile             3.113048
Agriculture         2.508227
Name: Occupation, dtype: float64

### People working in IT sector are ordering more follwed by Healthcare and aviation industry..

In [47]:
data['Product_Category'].value_counts(normalize=True) * 100 

Clothing & Apparel       23.596905
Food                     22.173797
Electronics & Gadgets    18.518189
Footwear & Shoes          9.463666
Household items           4.625100
Beauty                    3.753447
Games & Toys              3.433247
Sports Products           3.166415
Furniture                 3.139731
Pet Care                  1.885618
Office                    1.005070
Stationery                0.996175
Books                     0.916126
Auto                      0.880548
Decor                     0.853865
Veterinary                0.720448
Tupperware                0.640398
Hand & Power Tools        0.231255
Name: Product_Category, dtype: float64

### Highest orders belongs to Clothing & Apparel product category and 2nd most caegory is Food.. 

In [48]:
data[data["Amount"] == data['Amount'].max()]

User_ID  Cust_name Product_ID Gender Age Group  Age  Marital_Status  \
0  1002903  Sanskriti  P00125942      F     26-35   28               0   

         State     Zone  Occupation Product_Category  Orders   Amount  
0  Maharashtra  Western  Healthcare             Auto       1  23952.0

### highest order amount is 23952 from Maharashtra from healthcare sector and product category is Auto

In [49]:
data[data["Amount"] == data['Amount'].min()]

User_ID Cust_name Product_ID Gender Age Group  Age  Marital_Status  \
11250  1002744   Brumley  P00281742      F     18-25   19               0   

             State     Zone  Occupation Product_Category  Orders  Amount  
11250  Maharashtra  Western  Healthcare           Office       3   188.0

### lowest order amount is 188 from Maharashtra from healthcare sector and product category is Office

In [50]:
data.groupby("Product_Category").max()

User_ID  Cust_name Product_ID Gender Age Group  Age  \
Product_Category                                                             
Auto                   1005908     Zypern   P0096942      M       55+   83   
Beauty                 1006036     Zypern   P0099242      M       55+   89   
Books                  1005916  Wasserman   P0097742      M       55+   88   
Clothing & Apparel     1006039     percer   P0099342      M       55+   92   
Decor                  1005954     Yoseph   P0097242      M       55+   81   
Electronics & Gadgets  1006039     percer   P0099442      M       55+   91   
Food                   1006037     percer   P0099742      M       55+   92   
Footwear & Shoes       1006040     percer   P0099242      M       55+   92   
Furniture              1005982     percer   P0099242      M       55+   92   
Games & Toys           1006040        Zic   P0099042      M       55+   91   
Hand & Power Tools     1005780    Vasudev  P00358442      M       55+   85   
Household items        1006039     Zypern   P0099342      M       55+   88   
Office                 1006016    William   P0094942      M       55+   89   
Pet Care               1005954   Zrebassa   P0099042      M       55+   91   
Sports Products        1006040      Zydlo   P0099442      M       55+   89   
Stationery             1006010     Victor   P0097842      M       55+   87   
Tupperware             1006001     Yogesh   P0097842      M       55+   86   
Veterinary             1006014     Wilson   P0096542      M       55+   69   

                       Marital_Status          State     Zone Occupation  \
Product_Category                                                           
Auto                                1    Uttarakhand  Western    Textile   
Beauty                              1    Uttarakhand  Western    Textile   
Books                               1    Uttarakhand  Western    Textile   
Clothing & Apparel                  1    Uttarakhand  Western    Textile   
Decor                               1  Uttar Pradesh  Western    Textile   
Electronics & Gadgets               1    Uttarakhand  Western    Textile   
Food                                1    Uttarakhand  Western    Textile   
Footwear & Shoes                    1    Uttarakhand  Western    Textile   
Furniture                           1    Uttarakhand  Western    Textile   
Games & Toys                        1    Uttarakhand  Western    Textile   
Hand & Power Tools                  1  Uttar Pradesh  Western     Retail   
Household items                     1    Uttarakhand  Western    Textile   
Office                              1    Uttarakhand  Western    Textile   
Pet Care                            1    Uttarakhand  Western    Textile   
Sports Products                     1    Uttarakhand  Western    Textile   
Stationery                          1    Uttarakhand  Western    Textile   
Tupperware                          1    Uttarakhand  Western    Textile   
Veterinary                          1  Uttar Pradesh  Western     Retail   

                       Orders   Amount  
Product_Category                        
Auto                        4  23952.0  
Beauty                      4   7637.0  
Books                       4  13264.0  
Clothing & Apparel          4   8907.0  
Decor                       4  10072.0  
Electronics & Gadgets       4  10082.0  
Food                        4  19708.0  
Footwear & Shoes            4  20965.0  
Furniture                   4  20689.0  
Games & Toys                4  16499.0  
Hand & Power Tools          4  23434.0  
Household items             4   3900.0  
Office                      4    960.0  
Pet Care                    4   3555.0  
Sports Products             4  13702.0  
Stationery                  4  21563.0  
Tupperware                  4  21079.0  
Veterinary                  4   1777.0

### Though most of the orders(40%) are coming from age group 26-35 but maximum order value per categories are ordered by 55+ age group people..
### Though most of the orders(70%) are given by Women but All Highest order amount in each categories are belongs to Men only
### Most of the order are given by people having IT sector and Healthcare sector, Costliest products are purchased by people having occupations with Textile
### Most of the costliest products are ordered from Uttarakhand people. western zone are speding more amount

In [51]:
data.groupby("Occupation").max()

User_ID Cust_name Product_ID Gender Age Group  Age  \
Occupation                                                            
Agriculture      1006016    Zypern   P0099042      M       55+   91   
Automobile       1006036     liebe   P0099442      M       55+   90   
Aviation         1006040    percer   P0098442      M       55+   92   
Banking          1006036    percer   P0099342      M       55+   92   
Chemical         1006036    Zypern   P0099342      M       55+   92   
Construction     1006021    percer   P0099342      M       55+   92   
Food Processing  1006040    Zypern   P0099242      M       55+   87   
Govt             1006036     Zydlo   P0099742      M       55+   91   
Healthcare       1006036     liebe   P0099342      M       55+   92   
Hospitality      1006039    percer   P0098242      M       55+   90   
IT Sector        1006040     liebe   P0099442      M       55+   92   
Lawyer           1006039    percer   P0098742      M       55+   89   
Media            1006040     liebe   P0097242      M       55+   92   
Retail           1006016    Zypern   P0098242      M       55+   89   
Textile          1006036     Zydlo   P0098342      M       55+   89   

                 Marital_Status        State     Zone Product_Category  \
Occupation                                                               
Agriculture                   1  Uttarakhand  Western       Veterinary   
Automobile                    1  Uttarakhand  Western       Veterinary   
Aviation                      1  Uttarakhand  Western       Veterinary   
Banking                       1  Uttarakhand  Western       Veterinary   
Chemical                      1  Uttarakhand  Western       Veterinary   
Construction                  1  Uttarakhand  Western       Veterinary   
Food Processing               1  Uttarakhand  Western       Veterinary   
Govt                          1  Uttarakhand  Western       Veterinary   
Healthcare                    1  Uttarakhand  Western       Veterinary   
Hospitality                   1  Uttarakhand  Western       Veterinary   
IT Sector                     1  Uttarakhand  Western       Veterinary   
Lawyer                        1  Uttarakhand  Western       Veterinary   
Media                         1  Uttarakhand  Western       Veterinary   
Retail                        1  Uttarakhand  Western       Veterinary   
Textile                       1  Uttarakhand  Western       Tupperware   

                 Orders    Amount  
Occupation                         
Agriculture           4  23160.00  
Automobile            4  23924.00  
Aviation              4  23619.00  
Banking               4  23770.00  
Chemical              4  20955.00  
Construction          4  23912.00  
Food Processing       4  23877.00  
Govt                  4  23934.00  
Healthcare            4  23952.00  
Hospitality           4  23568.00  
IT Sector             4  23730.00  
Lawyer                4  23841.00  
Media                 4  23799.99  
Retail                4  23752.00  
Textile               4  21547.00

In [52]:
for cat in categories:
    print(cat)
    d = data[ data['Product_Category'] == cat and data["Amount"] == data['Amount'].max()]
    d

NameError: name 'categories' is not defined